java-frontend/src/main/java/org/sonar/java/cfg/CFG.java

```diff
@@ -1099,6 +1103,11 @@ private void buildInstanceOf(InstanceOfTree instanceOfTree) {
     build(instanceOfTree.expression());
   }
 
+  private void buildInstanceOf(PatternInstanceOfTree instanceOfTree) {
+    currentBlock.elements.add(instanceOfTree);
+    build(instanceOfTree.expression());
```

Comment:
what about the variable declaration?
```suggestion
    build(instanceOfTree.expression());
    build(instanceOfTree.variable());
```

Reply:
@m-g-sonar Do we do anything with variable declarations (that don't have an initializer) in the CFG?

Reply:
To answer my own question: When a variable declaration without an initializer is added to the CFG, the SE engine uses this to mark the declared variables as null. So since an `instance of`-declared variable will not be null, we should not add the variable declaration to the CFG. If anything, we should make the SE engine generate a not-null constraint when it sees an instance-of pattern, but that's outside of the scope of this PR.